core/sling/task_run_write.go

```diff
@@ -133,15 +134,229 @@ func (t *TaskExecution) WriteToFile(cfg *Config, df *iop.Dataflow) (cnt uint64,
+	}
+
+	// Pause dataflow to set up DDL and handlers
+	if paused := df.Pause(); !paused {
+		err = g.Error(err, "could not pause streams to infer columns")
+		return 0, err
+	}
+
+	// Prepare dataflow
+	sampleData, err := prepareDataflow(t, df, tgtConn)
+	if err != nil {
+		return 0, err
+	}
+
+	// Set table keys
+	tableTmp.Columns = sampleData.Columns
+	if err := tableTmp.SetKeys(cfg.Source.PrimaryKey(), cfg.Source.UpdateKey, cfg.Target.Options.TableKeys); err != nil {
+		err = g.Error(err, "could not set keys for "+tableTmp.FullName())
+		return 0, err
+	}
+
+	// Create temp table
+	if err := createTable(t, tgtConn, tableTmp, sampleData, true); err != nil {
+		err = g.Error(err, "could not create table "+tableTmp.FullName())
+		return 0, err
+	}
+
+	cfg.Target.Options.TableDDL = g.String(tableTmp.DDL)
+	cfg.Target.TmpTableCreated = true
+	df.Columns = sampleData.Columns
+	setStage("4 - load-into-temp")
+
+	// Add cleanup task for temp table
+	t.AddCleanupTaskFirst(func() {
+		if cast.ToBool(os.Getenv("SLING_KEEP_TEMP")) {
+			return
+		}
+
+		conn := tgtConn
+		if tgtConn.Context().Err() != nil {
+			conn, err = t.getTgtDBConn(context.Background())
+			if err == nil {
+				conn.Connect()
+			}
+		}
+		g.LogError(conn.DropTable(tableTmp.FullName()))
+		conn.Close()
+	})
+
+	// Begin transaction for temp table operations
+	if err := tgtConn.BeginContext(df.Context.Ctx); err != nil {
+		err = g.Error(err, "could not open transaction to write to temp table")
+		return 0, err
+	}
+
+	// Configure column handlers
+	if err := configureColumnHandlers(t, cfg, df, tgtConn, tableTmp); err != nil {
+		err = g.Error(err, "could not configure column handlers")
+		return 0, err
+	}
+
+	df.Unpause() // Resume dataflow
+	t.SetProgress("streaming data")
+
+	// Set batch limit if specified
+	if batchLimit := cfg.Target.Options.BatchLimit; batchLimit != nil {
+		df.SetBatchLimit(*batchLimit)
+	}
+
+	// Bulk import data into temp table
+	cnt, err = tgtConn.BulkImportFlow(tableTmp.FullName(), df)
+	if err != nil {
+		tgtConn.Rollback()
+		if cast.ToBool(os.Getenv("SLING_CLI")) && cfg.sourceIsFile() {
+			err = g.Error(err, "could not insert into %s.", tableTmp.FullName())
+		} else {
+			err = g.Error(err, "could not insert into "+tableTmp.FullName())
+		}
+		return 0, err
+	}
+
+	if err := tgtConn.Commit(); err != nil {
+		err = g.Error(err, "could not commit transaction")
+		return 0, err
+	}
+
+	t.PBar.Finish()
+
+	// Validate data
+	tCnt, err := tgtConn.GetCount(tableTmp.FullName())
+	if err != nil {
+		err = g.Error(err, "could not get count from temp table %s", tableTmp.FullName())
+		return 0, err
+	}
+	if cnt != tCnt {
+		err = g.Error(err, "inserted in temp table but table count (%d) != stream count (%d). Records missing/mismatch. Aborting", tCnt, cnt)
+		return 0, err
+	} else if tCnt == 0 && len(sampleData.Rows) > 0 {
+		err = g.Error(err, "Loaded 0 records while sample data has %d records. Exiting.", len(sampleData.Rows))
+		return 0, err
+	}
```

Comment:
this code is buggy, sorry about that, I find the `err` will pass to the internal, and got something like 
```
WRN NewError called with nil error:
```
old code
```
	tCnt, _ := tgtConn.GetCount(tableTmp.FullName())
	if cnt != tCnt {
		err = g.Error("inserted in temp table but table count (%d) != stream count (%d). Records missing/mismatch. Aborting", tCnt, cnt)
		return
	} else if tCnt == 0 && len(sampleData.Rows) > 0 {
		err = g.Error("Loaded 0 records while sample data has %d records. Exiting.", len(sampleData.Rows))
		return
	}
```